client/client.go

```diff
@@ -2099,10 +2103,10 @@ func (c *Client) AllocStateUpdated(alloc *structs.Allocation) {
 	stripped.DeploymentStatus = alloc.DeploymentStatus
 	stripped.NetworkStatus = alloc.NetworkStatus
 
-	select {
-	case c.allocUpdates <- stripped:
-	case <-c.shutdownCh:
-	}
+	c.allocUpdatesLock.Lock()
+	defer c.allocUpdatesLock.Unlock()
+
+	c.allocUpdates[stripped.ID] = stripped
```

Comment:
I think this was always true, but we can overwrite previous pending alloc updates for a given alloc, even if both updates were urgent priority? 

The `updates[alloc.ID] = alloc` would be the previous behavior. Just want to make sure.

Reply:
Right. The updates we get from the allocrunner are lightweight copies of its `*structs.Allocation` (i.e. just the client state), serialized with an internal lock. So whatever the latest update we've got here is the currently authoritative one. And the server doesn't keep versions of `*structs.Allocation` so it also only cares about the most recent version from the client.